inc/environment_indicator/namespace.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Altis\Cloud\Environment_Indicator;
+
+use function Altis\get_environment_name;
+use function Altis\get_environment_type;
+
+function bootstrap() {
+	add_action( 'admin_bar_init', __NAMESPACE__ . '\\enqueue_admin_scripts' );
+	add_action( 'admin_bar_menu', __NAMESPACE__ . '\\add_admin_bar_env_info' );
+}
+
+/**
+ * Add menu item to admin bar to display environment info:
+ * - environment type
+ * - environment name
+ * - URL to Altis dashboard for the environment
+ */
+function add_admin_bar_env_info() {
+	if ( ! is_admin_bar_showing() ) {
+		return;
+	}
+
+	// Specify capabilities for which environment indicator is shown in the admin bar.
+	$capability = apply_filters( 'altis_env_indicator_capability', 'manage_options' );
+	if ( ! current_user_can( $capability ) ) {
```

Comment:
Change of approach again here - this is really about differentiating the environments, at the same time we don't really want to affect production as it won't be relevant to most people using that.

Let's make it so that it always shows on `local`, `development` and `staging` by default, then filter that value with a hook called `altis.show_environment_indicator`. That way we can let our engineers decide on any complex logic need or want for this and we can consider incorporating it down the line.

```php
$show_indicator = in_array( get_environment_type(), [ 'local', 'development', 'staging' ], true );
$show_indicator = apply_filters( 'altis.show_environment_indicator', $show_indicator );
if ( ! $show_indicator ) {
     return;
}
```

Reply:
Could we instead allow returning false from `altis_env_indicator_text` and having that skip render?

Reply:
This is clearer IMO rather than overloading another filter and giving it multiple uses. I can't really think of another reason to filter the text that might not change the purpose of the feature.